vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -196,6 +210,42 @@ type internal ProjectInfoManager
         | true, (_loadTime, originalOptions) -> Some originalOptions
         | _ -> this.TryGetOptionsForProject(projectId) 
 
+    [<Export>]
+    member this.HandleCommandLineChanges(path:string, sources:ImmutableArray<CommandLineSourceFile>, references:ImmutableArray<CommandLineReference>) =
+        let projectDir = Path.GetDirectoryName(path)
+        let fullPath p = if Path.IsPathRooted(p) then p else Path.Combine(projectDir, p)
+        let sourcePaths = sources |> Seq.map(fun s -> fullPath s.Path) |> Seq.toArray
+        let referencePaths = references |> Seq.map(fun r -> fullPath r.Reference) |> Seq.toArray
+
+        LogFile.log "======================================="
+        LogFile.log path 
+        LogFile.log projectDir
+        LogFile.log "======================================="
+        sourcePaths |> Seq.iter(fun s -> LogFile.log s)
+        LogFile.log "======================================="
+        referencePaths |> Seq.iter(fun r -> LogFile.log r)
+        LogFile.log "======================================="
+
+        lock lockObject (fun () -> match sourceFiles.TryGetValue path with
+                                   | true, _ -> sourceFiles.Remove(path) |> ignore
+                                                sourceFiles.Add(path, sourcePaths)
+                                   | _       -> sourceFiles.Add(path, sourcePaths)
+                                   match referenceFiles.TryGetValue path with
+                                   | true, _ -> referenceFiles.Remove(path) |> ignore
+                                                referenceFiles.Add(path, referencePaths)
+                                   | _       -> referenceFiles.Add(path, referencePaths) )
+        ()
+
+    member __.GetSourceFiles(path:string) = 
+        lock lockObject (fun () -> match sourceFiles.TryGetValue path with
+                                   | true, value -> value
+                                   | _           -> Array.empty<string>)
```

Comment:
Normal F# practice is to use ``[| |]`` for ``Array.empty``, or to use just ``Array.empty`` without the type annotation (which is inferred)